packages/strapi-plugin-users-permissions/services/Providers.js

```diff
@@ -52,18 +52,19 @@ exports.connect = (provider, query) => {
       });
     } else {
       // Get the profile.
-      getProfile(provider, query, (err, profile) => {
+      getProfile(provider, query, async (err, profile) => {
         if (err) {
           reject(err);
```

Comment:
You could use `return reject(err)` pattern to avoid indentation and the else part. This pattern could be repeated two or three times and this part of the code will be flattened (more readable).